sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -308,7 +318,7 @@ static long getEstimatedSizeBytes(Datastore datastore, Query query, @Nullable St
       LOG.info("Latest stats timestamp for kind {} is {}", ourKind, latestTimestamp);
 
       Query.Builder queryBuilder = Query.newBuilder();
-      if (namespace == null) {
+      if (namespace == null || namespace.isEmpty()) {
         queryBuilder.addKindBuilder().setName("__Stat_Kind__");
```

Comment:
Why is this correct? It seems like (at runtime) the namespace may be either null or not null, and we'd need to change the query based on that. Add a comment?